pkg/agent/core/ngt/handler/grpc/handler_test.go

```diff
@@ -3972,90 +3972,900 @@ func Test_server_MultiInsert(t *testing.T) {
-		wantRes *payload.Object_Location
-		err     error
+		code     codes.Code
+		wantUuid string
 	}
 	type test struct {
 		name       string
 		args       args
-		fields     fields
 		want       want
 		checkFunc  func(want, *payload.Object_Location, error) error
-		beforeFunc func(args)
+		beforeFunc func(args) (Server, error)
 		afterFunc  func(args)
 	}
 	defaultCheckFunc := func(w want, gotRes *payload.Object_Location, err error) error {
-		if !errors.Is(err, w.err) {
-			return errors.Errorf("got_error: \"%#v\",\n\t\t\t\twant: \"%#v\"", err, w.err)
-		}
-		if !reflect.DeepEqual(gotRes, w.wantRes) {
-			return errors.Errorf("got: \"%#v\",\n\t\t\t\twant: \"%#v\"", gotRes, w.wantRes)
+		if err != nil {
+			st, ok := status.FromError(err)
+			if !ok {
+				errors.Errorf("got error cannot convert to Status: \"%#v\"", err)
+			}
+			if st.Code() != w.code {
+				return errors.Errorf("got code: \"%#v\",\n\t\t\t\twant code: \"%#v\"", st.Code(), w.code)
+			}
+		} else {
+			if uuid := gotRes.GetUuid(); w.wantUuid != uuid {
+				return errors.Errorf("got uuid: \"%#v\",\n\t\t\t\twant uuid: \"%#v\"", uuid, w.wantUuid)
+			}
 		}
 		return nil
 	}
+
+	const (
+		insertNum = 1000
+		dimension = 128
+	)
+
+	defaultUpdateConfig := &payload.Update_Config{
+		SkipStrictExistCheck: true,
+	}
+	toFloat32Vector := func(v []uint8) (ret []float32) {
+		ret = make([]float32, 0, len(v))
+		for _, e := range v {
+			ret = append(ret, float32(e))
+		}
+		return
+	}
+	beforeFunc := func(objectType string) func(args) (Server, error) {
+		cfg := &config.NGT{
+			Dimension:        dimension,
+			DistanceType:     ngt.L2.String(),
+			ObjectType:       objectType,
+			CreationEdgeSize: 60,
+			SearchEdgeSize:   20,
+			KVSDB: &config.KVSDB{
+				Concurrency: 10,
+			},
+			VQueue: &config.VQueue{
+				InsertBufferPoolSize: 1000,
+				DeleteBufferPoolSize: 1000,
+			},
+		}
+		var gen func(int, int) [][]float32
+		switch objectType {
+		case ngt.Float.String():
+			gen = vector.GaussianDistributedFloat32VectorGenerator
+		case ngt.Uint8.String():
+			gen = func(n, dim int) (ret [][]float32) {
+				vectors := vector.GaussianDistributedUint8VectorGenerator(n, dim)
+				ret = make([][]float32, 0, len(vectors))
+				for _, vector := range vectors {
+					ret = append(ret, toFloat32Vector(vector))
+				}
+				return
+			}
+		}
+
+		return func(a args) (Server, error) {
+			eg, ctx := errgroup.New(a.ctx)
+			ngt, err := service.New(cfg, service.WithErrGroup(eg), service.WithEnableInMemoryMode(true))
+			if err != nil {
+				return nil, err
+			}
+
+			s, err := New(WithErrGroup(eg), WithNGT(ngt))
+			if err != nil {
+				return nil, err
+			}
+
+			reqs := make([]*payload.Insert_Request, insertNum)
+			for i, v := range gen(insertNum, cfg.Dimension) {
+				reqs[i] = &payload.Insert_Request{
+					Vector: &payload.Object_Vector{
+						Id:     strconv.Itoa(i),
+						Vector: v,
+					},
+					Config: &payload.Insert_Config{
+						SkipStrictExistCheck: true,
+					},
+				}
+			}
```

Comment:
I suppose we can use `genMultiInsertReq` function in https://github.com/vdaas/vald/pull/1612
could you please add `TODO` comment?
```suggestion
                        // TODO: use genMultiInsertReq()
			reqs := make([]*payload.Insert_Request, insertNum)
			for i, v := range gen(insertNum, cfg.Dimension) {
				reqs[i] = &payload.Insert_Request{
					Vector: &payload.Object_Vector{
						Id:     strconv.Itoa(i),
						Vector: v,
					},
					Config: &payload.Insert_Config{
						SkipStrictExistCheck: true,
					},
				}
			}
```